plugin/pkg/scheduler/factory/factory.go

```diff
@@ -112,21 +116,47 @@ func NewConfigFactory(client *client.Client, schedulerName string) *ConfigFactor
 		0,
 		framework.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
-				if pod, ok := obj.(*api.Pod); ok {
-					c.modeler.LockedAction(func() {
-						c.modeler.ForgetPod(pod)
-					})
+				pod, ok := obj.(*api.Pod)
+				if !ok {
+					panic("cannot convert to *api.Pod")
```

Comment:
I don't think we should have panic's in the code - probably glog.Errorf() + return